### ISSUE TITLE
Regex not cooperating

### ISSUE THREAD

**Issue by johnsickels** [flocasts/flagpole#58]
I want to use regex in `.find()` but it is being ignored. It clicks the first matching element instead, regardless of text inside. 

Repro:
```
suite
  .scenario("Click on History", "browser")
  .open("https://en.wikipedia.org/wiki/Regular_expression")
  .next(async (context) => {
    // It finds "Patterns" instead, the first `#toc a`
    const history = await context.find("#toc a", /^Hist/);
    scenarioB.open(history);
  });
const scenarioB = suite
  .scenario("Should be at History", "browser")
  .next(async (context) => {
    context
      .assert(context.response.url)
      .equals("https://en.wikipedia.org/wiki/Regular_expression#History");
  });
```

I also tried all flavors, like `/^Hist.*$/`, etc.. am I doing it incorrectly?

Flagpole version: 2.5.18

**Comment by johnsickels** [comment]
My true case looks like this
```
await streamAScenario.open(await context.find("h3 a", /Stream A/));
```

**Comment by johnsickels** [comment]
I think I need to specify this is for a browser scenario

**Comment by jasonbyrne** [comment]
I can reproduce this issue, looking into it now

**Comment by jasonbyrne** [comment]
Fixed with upcoming v2.5.19
